mongo/data_store.go

```diff
@@ -166,6 +166,66 @@ func (m *JobStore) GetTask(ctx context.Context, jobID string, taskName string) (
 	return task, err
 }
 
+// GetTasks retrieves all the tasks, from the collection, and lists them in order of last_updated
+func (m *JobStore) GetTasks(ctx context.Context, offsetParam string, limitParam string, jobID string) (models.Tasks, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+	log.Info(ctx, "getting list of tasks")
+	results := models.Tasks{}
+
+	// If an empty jobID or taskName was passed in, return an error with a message.
+	if jobID == "" {
+		return results, ErrEmptyIDProvided
+	}
+
+	_, err := m.findJob(s, jobID, models.Job{})
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return results, ErrJobNotFound
+		}
+		return results, err
+	}
+
+	numTasks, _ := s.DB(m.Database).C(m.TasksCollection).Find(bson.M{"job_id": jobID}).Count()
```

Comment:
We should still handle this error here, we wouldn't want the code to panic in case we lose the connection to mongo.

Reply:
makes sense - done